packages/transport/src/modules/sync/SyncController.ts

```diff
@@ -404,21 +405,48 @@ export class SyncController extends TransportController {
 
         const externalEvents = await getExternalEventsResult.value.collect();
 
-        const externalEventProcessor = new ExternalEventsProcessor(
-            this.parent.messages,
-            this.parent.relationships,
-            externalEvents,
-            reporter,
-            this.eventBus,
-            this.parent.identity.address.toString()
-        );
-        await externalEventProcessor.execute();
+        const syncStep = reporter.createStep(SyncStep.ExternalEventsProcessing, externalEvents.length);
+        const results: FinalizeSyncRunRequestExternalEventResult[] = [];
+        const changedItems = new ChangedItems();
+
+        for (const externalEvent of externalEvents) {
+            try {
+                const externalEventProcessorConstructor = this.externalEventRegistry.getProcessorForItem(externalEvent.type);
+                const item = await new externalEventProcessorConstructor(this.eventBus, changedItems, this.parent).execute(externalEvent);
+
+                if (item) {
+                    changedItems.addItem(item);
+                }
```

Comment:
I like if short things like this are one-liners (no change required if you feel different)